### PR TITLE
migrate to the new build-and-push-action

### PR DESCRIPTION
**Pull request by zmiklank**

### CODE REVIEW

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -8,54 +8,17 @@ jobs:
...
-          image: ${{ steps.build-image.outputs.image }}
-          tags: ${{ steps.build-image.outputs.tags }}
-          registry: quay.io/centos7
-          username: ${{ secrets.QUAY_IMAGE_BUILDER_USERNAME }}
-          password: ${{ secrets.QUAY_IMAGE_BUILDER_TOKEN }}
-
-      - name: Print image url
-        run: echo "Image pushed to ${{ steps.push-to-quay.outputs.registry-paths }}"
+          registry: "quay.io"
+          registry_namespace: "centos7"
```
Comment (phracek) [inline .github/workflows/build-and-push.yml:19]: What about C9S? This repository belongs to sclorg.

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -8,54 +8,17 @@ jobs:
...
+          registry_username: ${{ secrets.QUAY_IMAGE_BUILDER_USERNAME }}
+          registry_token: ${{ secrets.QUAY_IMAGE_BUILDER_TOKEN }}
+          docker_context: ${{ matrix.dockerfile_path }}
+          dockerfile_path: ${{ matrix.dockerfile_path }}
+          tag: "centos7"
```
Comment (phracek) [inline .github/workflows/build-and-push.yml:24]: The same here. It should be mentioned in the matrix.

**Review by phracek — CHANGES_REQUESTED** [review]
Thanks for the PR! Please address also my changes.

File: .github/workflows/build-and-push.yml (hunk 1 above)
Comment (zmiklank) [reply .github/workflows/build-and-push.yml:19]: Thanks for the review.
I only rewrote the action as it was. I didn't want to add any new functionality. 

I would suggest creating other PR for adding c9s support, to test it separately, but if you really want, I can add it here, WDYT?

File: .github/workflows/build-and-push.yml [hunk 3]
```diff
@@ -8,54 +8,17 @@ jobs:
   build-and-push:
     runs-on: ubuntu-latest
     strategy:
```
Comment (phracek) [inline .github/workflows/build-and-push.yml:10]:
```suggestion
    if: github.repository_owner == 'sclorg'
    strategy:
```

**Review by phracek — CHANGES_REQUESTED** [review]
Thanks for the PR!. Look at my proposals.

File: .github/workflows/build-and-push.yml [hunk 4]
```diff
@@ -7,55 +7,19 @@ on:
 jobs:
   build-and-push:
     runs-on: ubuntu-latest
+    if: github.repository_owner == 'sclorg'
     strategy:
+      fail-fast: false
       matrix:
-        version: [14, 16]
+        dockerfile_path: [ "14", "16" ]
```
Comment (phracek) [inline .github/workflows/build-and-push.yml:14]: And what bout minimal?
As far as I know, we supports `12 14 14-minimal 16 16-minimal`

**Review by phracek — CHANGES_REQUESTED** [review]
Thanks for the PR!. Please address my changes.

File: .github/workflows/build-and-push.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (zmiklank) [reply .github/workflows/build-and-push.yml:11]: This is already included.

File: .github/workflows/build-and-push.yml (hunk 4 above)
Comment (zmiklank) [reply .github/workflows/build-and-push.yml:14]: They have their own action, but I can check and compare, and add them here if possible.

**Pull request merged**